standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5158,20 +5157,27 @@ public DropPartitionsResult drop_partitions_req(
               }
             }
           }
+          if (result.isEmpty()) {
+            hasMissingParts = true;
+            if (!ifExists) {
+              // fail-fast for missing partition expr
+              break;
+            }
+          }
           parts.addAll(result);
         }
       } else if (spec.isSetNames()) {
         partNames = spec.getNames();
-        minCount = partNames.size();
         parts = ms.getPartitionsByNames(catName, dbName, tblName,
             new GetPartitionsArgs.GetPartitionsArgsBuilder()
                 .partNames(partNames).skipColumnSchemaForPartition(request.isSkipColumnSchemaForPartition())
                 .build());
+        hasMissingParts = (parts.size() != partNames.size());
```

Comment:
Addressed, also add the `dropPartitionsByNames` api in Hive client.

Reply:
https://github.com/apache/hive/pull/4905/commits/7870dadeca8f73ae92619cc0c0014cecba0c689e You added the `dropPartitionsByNames` api, does it mean we never use this `else if` clause before?
If so, please revert this commit https://github.com/apache/hive/pull/4905/commits/7870dadeca8f73ae92619cc0c0014cecba0c689e, we don't need add a redundant new api to test it.

Reply:
This new api is meaningful for `Hive.java` user, I think it's not redundant even though it's not related to this patch. 
It's fine to raise another PR for this change if necessary. also cc: @aturoczy  @deniskuzZ @ayushtkn

Reply:
@wecharyu I think we need submit another PR to add the new interface instead of this PR. If the new interface is meaningful , i think this will attract more developers attention. Thx.

Reply:
Make sense. Reverted it.